src/components/modebar/modebar.js

```diff
@@ -45,13 +45,28 @@ var proto = ModeBar.prototype;
 proto.update = function(graphInfo, buttons) {
     this.graphInfo = graphInfo;
 
-    var context = this.graphInfo._context;
+    var context = this.graphInfo._context,
+        fullLayout = this.graphInfo._fullLayout,
+        modeBarId = 'modebar-' + fullLayout._uid;
+
+    this.element.setAttribute('id', modeBarId);
+    this._uid = modeBarId;
 
     if(context.displayModeBar === 'hover') {
         this.element.className = 'modebar modebar--hover';
     }
     else this.element.className = 'modebar';
 
+    if(fullLayout.modebar.orientation === 'v') {
+        this.element.className += ' vertical';
+        buttons = buttons.reverse();
+    }
+
+    Lib.addRelatedStyleRule(modeBarId, '#' + modeBarId, 'background-color: ' + fullLayout.modebar.bgcolor);
```

Comment:
Do we _have_ to use CSS here? Could we instead add `mouseover` / `mouseout` handlers to the modebar buttons, similar to how we handle updatemenu buttons:

https://github.com/plotly/plotly.js/blob/30ed4a4a59789a486b70dda360d630d290811816/src/components/updatemenus/draw.js#L219-L221

https://github.com/plotly/plotly.js/blob/30ed4a4a59789a486b70dda360d630d290811816/src/components/updatemenus/draw.js#L464-L467

that way we wouldn't have to hijack the `<style>` dynamically.

Reply:
> Since we already add/remove class name `active` to the icons

Right, getting "active" to work would require some work. The modebar button click handlers would need to update their style attribute instead of appending/removing the `"active"` class name. 

Consider my comment non-blocking.